frontend/src/app/services/application.service.ts

```diff
@@ -55,6 +57,21 @@ export class ApplicationService {
       distinctUntilKeyChanged("id"),
       switchMap((application: Application) => this.updateCsiForPages(application))
     ).subscribe(this.pageCsis$);
+
+    this.getFailingJobs().pipe(
+      map(failingJobs => {
+        if (failingJobs) {
+          return failingJobs.reduce((failingJobsByApplication, currentValue) => {
```

Comment:
Could you extract the return code into an extra function (something like return reduceFailedJobs())